src/stream/src/executor/dynamic_filter.rs

```diff
@@ -327,14 +327,15 @@ impl<S: StateStore> DynamicFilterExecutor<S> {
                     // Flush the difference between the `prev_value` and `current_value`
                     let curr: Datum = current_epoch_value.clone().flatten();
                     let prev: Datum = prev_epoch_value.flatten();
+                    let row_deserializer = RowDeserializer::new(self.schema.data_types());
                     if prev != curr {
                         let (range, latest_is_lower, is_insert) = self.get_range(&curr, prev);
                         for (_, rows) in self.range_cache.range(range, latest_is_lower) {
                             for row in rows {
                                 if let Some(chunk) = stream_chunk_builder.append_row_matched(
                                     // All rows have a single identity at this point
                                     if is_insert { Op::Insert } else { Op::Delete },
-                                    row,
+                                    &row_deserializer.deserialize(row.row.as_ref())?,
                                 )? {
```

Comment:
we need add interface on StreamChunkBuilder accepting CompactRow

Reply:
Well, it will need to deserialize Vec<u8> into buffer of sorts anyway. So I think this option is not that bad.

Are you saying we should use this interface? Seems complicated. How to get the `ArrayImpl` from Vec<u8> and datatypes?https://github.com/risingwavelabs/risingwave/blob/6ec1948a3e35785b3a4f9bb20d8c1f7ff4ed9504/src/common/src/array/mod.rs#L416